statescript/executor.go

```diff
@@ -179,8 +210,43 @@ func execute(name string, timeout time.Duration) int {
 	return 0
 }
 
+// Catches a script that requests a retry in a loop. Is limited by the total window given to a script demanding a
+// retry.
+func executeScript(s os.FileInfo, dir string, l Launcher, timeout time.Duration, ignoreError bool) error {
+
+	iet := time.Now()
+	for {
+		switch ret := execute(filepath.Join(dir, s.Name()), timeout); ret {
+		case 0:
+			// success
+			return nil
+		case exitRetryLater:
+			if ignoreError {
+				log.Errorf("statescript: ignoring error executing '%s': %d", s.Name(), ret)
+				return nil
+			}
+			if time.Since(iet) <= l.getRetryTimeout() {
+				log.Errorf("statescript: %s requested a retry", s.Name())
+				time.Sleep(l.getRetryInterval())
+				continue
+			}
+			return errors.New("statescript: retry time-limit exceeded")
+		default:
+			// In case of error scripts all should be executed.
+			if ignoreError {
+				log.Errorf("statescript: ignoring error executing '%s': %d", s.Name(), ret)
+				return nil
+			}
+			return errors.Errorf("statescript: error executing '%s': %d",
+				s.Name(), ret)
+		}
+	}
+}
+
 func (l Launcher) ExecuteAll(state, action string, ignoreError bool) error {
+
```

Comment:
Not a big thing, but I would rather avoid adding those empty lines every now and then in the existing code. I don't care about the new code, but if there are lot of those small not important changes it is harder to follow the changes later (when using git blame for example). Just a hint for future.